### ISSUE TITLE
Extend "jsx-boolean-value" to support optional "ignore" list

### ISSUE THREAD

**Issue by mehdivk** [yannickcr/eslint-plugin-react#1249]
It would be great if we could configure `jsx-boolean-value` to ignore a list of props.

### Example
```js
{
  'react/jsx-boolean-value': ['error', 'never', { ignore: ['value']}]
}
```

This allows the following expression to be valid

```jsx
<Toggle value={true} />
```

And if you wonder why I'm not using a prop like `toggled` instead of `value`, it's because we want to have a set of components that support the same prop.

I'm more than happy to support the idea with a PR.

**Comment by ljharb** [comment]
why wouldn't you want to do `<Toggle value />`?

**Comment by mehdivk** [comment]
> why wouldn't you want to do <Toggle value />?

Good question. I think `<Toggle value />` has some readability issues. For props that ends with `ed` (E.g. checked, toggled, disabled etc) it makes sense to drop `{true}` but not for something like `value`.

**Comment by ljharb** [comment]
It sounds like then you wouldn't want "ignore", you'd instead want a setting that enforced "always" for that prop.

**Comment by mehdivk** [comment]
> It sounds like then you wouldn't want "ignore", you'd instead want a setting that enforced "always" for that prop.

Yes, I want to use a mix of `always` and `never` in my project. 

This should cause lint error
```jsx
<Toggle toggled={true} />
```

but this should pass the linting
```jsx
<CheckBox value={true} />
```

and this should cause lint error
```jsx
<CheckBox value />
```

**Comment by ljharb** [comment]
Although I don't personally agree with that style, an option to the rule to allow this works.

What about, only with "always", a "never" option - and only with "never", an "always" option? Both options would be a unique array of prop names.

**Comment by timhwang21** [comment]
Timothy Hwang liked your email
Spark by Readdle

**Comment by ljharb** [comment]
PR up: #1250

**Comment by mehdivk** [comment]
Thanks for the PR @ljharb
